From 286a20396a78a4ebebb132dfd71ba23377b884a5 Mon Sep 17 00:00:00 2001
From: Daniel Charles <daniel.charles@intel.com>
Date: Wed, 9 May 2012 11:28:37 -0700
Subject: Change max width and height supported by H.263 decoder.

Currently this is set to CIF PAL resolution (352x288) which
won't calculate the actual output buffer size for higher resolutions

The h.263 decoder needs the max width and height supported to be able
to decode

H.263 decoders supports: QCIF, CIF, D1 and 16CIF; all PAL.

Change-Id: If5e44c522cb3cb56bb6235ec2992e2544b1737e9
Original-Change-Id: I7f8d2ae4263925ac4cf9250404ef92fc819ca33a
Signed-off-by: Daniel Charles <daniel.charles@intel.com>
---
 .../codecs/m4v_h263/dec/SoftMPEG4.cpp              |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/codecs/m4v_h263/dec/SoftMPEG4.cpp b/media/libstagefright/codecs/m4v_h263/dec/SoftMPEG4.cpp
index 020cc0a..dc55fb5 100644
--- a/media/libstagefright/codecs/m4v_h263/dec/SoftMPEG4.cpp
+++ b/media/libstagefright/codecs/m4v_h263/dec/SoftMPEG4.cpp
@@ -116,6 +116,13 @@ void SoftMPEG4::initPorts() {
             ? const_cast<char *>(MEDIA_MIMETYPE_VIDEO_MPEG4)
             : const_cast<char *>(MEDIA_MIMETYPE_VIDEO_H263);
 
+    if (mMode == MODE_H263) {
+        // H.263 requires the max size supported by the decoder
+        // which is 16CIF (1408x1152)
+        mWidth=1408;
+        mHeight=1152;
+    }
+
     def.format.video.pNativeRender = NULL;
     def.format.video.nFrameWidth = mWidth;
     def.format.video.nFrameHeight = mHeight;
-- 
1.7.4.1

